thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -816,11 +853,19 @@ class CDistributorBase : implements IHashDistributor, implements IExceptionHandl
                     const void *row = input->ungroupedNextRow();
                     if (!row)
                         break;
-                    unsigned dest = owner.ihash->hash(row)%owner.numnodes;
-                    CTarget *target = targets.item(dest);
-                    if (target->getSenderFinished())
-                        ReleaseThorRow(row);
+
+                    CTarget *target = nullptr;
+                    if (owner.isAll)
+                        target = targets.item(0);
```

Comment:
Should this be checking getSenderFinished(0)?

Reply:
No, intentionally was not checking any finished status here.
In the single standard target case, it checks so it can throw the row away early if target has flagged that it's done.
Instead it checks to see if individual targets are finished during the broadcast (sendToOthers)